tests/accessor/accessor_api_image_common.h

```diff
@@ -984,10 +984,28 @@ class image_accessor_api_sampled_r {
        *    ftp://ftp.icsi.berkeley.edu/pub/theory/priest-thesis.ps.Z
        *  for details
        */
-      return get_expected_value_linear(idx + 1);
+      return get_expected_value_linear(next_id(idx));
     }
   }
 
+  /**
+   *  @brief Computes the next id for getting an expected upper value for the
+   *  linear filtration. If the target is target::image_array, the last
+   *  component is not a coordinate, it is the index of an image in the array
+   *  to read from, so the last component should not be incremented.
+   */
+  template <int dims>
+  sycl::id<dims> next_id(sycl::id<dims> idx) const {
+    if constexpr (target == sycl::target::image_array) {
+      auto res = idx;
+      for (int i = 0; i + 1 < dims; i++) {
+        res[i]++;
+      }
+      return res;
+    }
+    return idx + 1;
```

Comment:
```suggestion
    static_assert(dims > 0);
    ++idx;
    if constexpr (target == sycl::target::image_array) {
      --idx[dims-1];
    }
    return idx;
```
Just as a suggestion. We are not going to wrap on `size_t` anyway, as it would require quite huge images. Assert is going to have no effect for SYCL-2020 rev.4

Reply:
@yuriykoch Thank you for the suggestion, I've committed it.